packages/block-library/src/search/index.php

```diff
@@ -360,13 +378,118 @@ function styles_for_block_core_search( $attributes ) {
 		$button_styles[] = sprintf( 'background: %s;', $attributes['style']['color']['gradient'] );
 	}
 
+	// Add typography styles.
+	$has_font_size = ! empty( $attributes['style']['typography']['fontSize'] );
```

Comment:
Note to self: this is possibly something the style engine could optimize, since it will do all the `$has_` checks internally, and return the classnames. 

E.g., 

```php
$typography_styles = gutenberg_style_engine_get_styles( array( 'typography' => $$attributes['style']['typography'] ) );
```
$button_styles = $typography_styles['css'];
$input_styles = $typography_styles['css'];

if ( $show_label ) {
    $label_styles = $typography_styles['css'];
}
```

Doesn't take into account no textDecoration for the input. 

Maybe a follow up refactor anyway....